### PR TITLE
add `pre-commit` setup with black and sort

### PR DESCRIPTION
**Pull request by sbdchd**
Some more setup for https://github.com/scrapinghub/extruct/issues/199

once merged we can add the commit sha to the ignored revs file

### CODE REVIEW

File: extruct/w3cmicrodata.py [hunk 1]
```diff
@@ -46,13 +45,19 @@
 
 class LxmlMicrodataExtractor(object):
     # iterate in document order (used below for fast get_docid)
-    _xp_item = lxml.etree.XPath('descendant-or-self::*[@itemscope]')
-    _xp_prop = lxml.etree.XPath("""set:difference(.//*[@itemprop],
+    _xp_item = lxml.etree.XPath("descendant-or-self::*[@itemscope]")
+    _xp_prop = lxml.etree.XPath(
+        """set:difference(.//*[@itemprop],
                                                   .//*[@itemscope]//*[@itemprop])""",
```
Comment (lopuhin) [inline extruct/w3cmicrodata.py:51]: that bit looks a bit weird, I wonder if there is a way of editing it so that black is happy and it looks decent?

**Review by lopuhin — COMMENTED** [review]
Thanks @sbdchd , I checked the formatting changes and have only one comment below. Do you mind checking the build failures? Also it would be great to add something to the bottom of the README - telling that we use black and isort, and documenting any development setup, if needed.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+repos:
+  - hooks:
+      - id: black
+        language_version: python3
+    repo: https://github.com/ambv/black
+    rev: 22.3.0
```
Comment (BurnzZ) [inline .pre-commit-config.yaml:6]: Let's use the latest `22.10.0` version while we're at it.

File: extruct/_extruct.py [hunk 3]
```diff
@@ -1,169 +1,185 @@
...
+    encoding="UTF-8",
+    syntaxes=SYNTAXES,
+    errors="strict",
+    uniform=False,
+    return_html_node=False,
+    schema_context="http://schema.org",
+    with_og_array=False,
+    **kwargs
+):
     """htmlstring: string with valid html document;
```
Comment (BurnzZ) [inline extruct/_extruct.py:29]: This looks a bit strange as it's not aligned with the rest of the params below. We can move this to the new line.

File: extruct/tool.py [hunk 4]
```diff
@@ -1,53 +1,75 @@
...
+    arg(
+        "--schema_context",
+        default="http://schema.org",
+        help="schema's context for current page",
+    )
+    arg(
+        "--errors",
+        default="log",
+        choices=["strict", "log", "ignore"],
         help="errors: set to 'log'(default) to log the exceptions, 'ignore' to ignore"
```
Comment (BurnzZ) [inline extruct/tool.py:68]: Looks a bit strange. I'm curious as to why black didn't format this one. Generally, for long strings, we'd do:
```python
help=(
    "errors: set to 'log'(default) to log the exceptions, 'ignore' to ignore "
    "them or 'strict' to raise them"
)
```

File: .pre-commit-config.yaml (hunk 2 above)
Comment (sbdchd) [inline .pre-commit-config.yaml:6]:
```suggestion
    rev: 22.10.0
```

File: extruct/_extruct.py (hunk 3 above)
Comment (sbdchd) [inline extruct/_extruct.py:29]:
```suggestion
    """
    htmlstring: string with valid html document;
```

File: extruct/tool.py (hunk 4 above)
Comment (sbdchd) [reply extruct/tool.py:68]: I think it's because it's already split into two strings

File: .github/workflows/python-package.yml [hunk 5]
```diff
@@ -30,4 +30,26 @@ jobs:
         python -m pip install -r requirements-dev.txt
     - name: tox
       run: |
-        tox -e `python -c "import sys; print('py' + ''.join(sys.version.split('.')[:2]))"`
+        tox -e ${{ matrix.python-version }}
```
Comment (sbdchd) [inline .github/workflows/python-package.yml:33]:
```suggestion
        tox -e `python -c "import sys; print('py' + ''.join(sys.version.split('.')[:2]))"`
```

File: .github/workflows/python-package.yml [hunk 6]
```diff
@@ -10,13 +10,13 @@ on:
...
 jobs:
-  build:
+  test:
 
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        python-version: ['3.6', '3.7', '3.8', '3.9']
+        python-version: ['py36', 'py37', 'py38', 'py39']
```
Comment (sbdchd) [inline .github/workflows/python-package.yml:19]:
```suggestion
        python-version: ['3.6', '3.7', '3.8', '3.9']
```

**Review by lopuhin — APPROVED** [review]
Thanks @sbdchd looks good to me 👍 Let's also wait for @BurnzZ to have another look.

File: extruct/tool.py (hunk 4 above)
Comment (BurnzZ) [reply extruct/tool.py:68]: Let's leave it as-is for now.

**Review by BurnzZ — APPROVED** [review]
LGTM, thanks @sbdchd !

**Comment by BurnzZ** [comment]
Took the liberty of creating the `.git-blame-ignore-revs` in https://github.com/scrapinghub/extruct/pull/202.

**Pull request merged**